public/apps/configuration/panels/audit-logging/audit-logging-general-settings.tsx

```diff
@@ -0,0 +1,162 @@
+  EuiPageHeader,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { cloneDeep, set } from 'lodash';
+import { Toast } from '@elastic/eui/src/components/toast/global_toast_list';
+import { AppDependencies } from '../../../types';
+import { EditSettingGroup } from './edit-setting-group';
+import { SETTING_GROUPS } from './constants';
+import { API_ENDPOINT_AUDITLOGGING } from '../../constants';
+import { updateAuditLogging } from '../../utils/audit-logging-view-utils';
+import { AuditLoggingSettings } from './types';
+
+export function AuditLoggingEditGeneralSetting(props: AppDependencies) {
+  const [config, setConfig] = useState<AuditLoggingSettings>({});
+  const [editConfig, setEditConfig] = useState<AuditLoggingSettings>({});
+
+  const [toasts, setToasts] = useState<Toast[]>([]);
+  const addToast = useCallback((toastToAdd: Toast) => {
+    setToasts((state) => state.concat(toastToAdd));
+  }, []);
+  const removeToast = (toastToDelete: Toast) => {
+    setToasts(toasts.filter((toast) => toast.id !== toastToDelete.id));
+  };
+
+  const handleChange = (setting, val) => {
+    const editedConfig = set(cloneDeep(editConfig), setting.path, val);
+
+    setEditConfig(editedConfig);
+  };
+
+  useEffect(() => {
+    const fetchConfig = async () => {
+      try {
+        const rawConfig = await props.coreStart.http.get(API_ENDPOINT_AUDITLOGGING);
+        const fetchedConfig = rawConfig.data.config;
+        setConfig(fetchedConfig);
+        setEditConfig(fetchedConfig);
+      } catch (e) {
+        console.log(e);
+      }
+    };
+
+    fetchConfig();
+  }, [props.coreStart.http]);
+
+  const renderSave = () => {
+    return (
+      <>
+        <EuiFlexGroup justifyContent="flexEnd">
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              onClick={() => {
+                cancel();
+              }}
+            >
+              Cancel
+            </EuiButton>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              fill
+              onClick={() => {
+                saveConfig(editConfig);
+              }}
+            >
+              Save
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </>
+    );
+  };
+
+  const cancel = () => {
+    window.scrollTo({ top: 0 });
+    setEditConfig(config);
+  };
+
+  const saveConfig = async (configToUpdate: object) => {
+    try {
+      await updateAuditLogging(props.coreStart.http, configToUpdate);
+
+      const successToast: Toast = {
+        id: 'update-result',
+        color: 'success',
+        iconType: 'check',
+        title: `Audit configuration was successfully updated.`,
+      };
+
+      addToast(successToast);
+      setConfig(editConfig);
+    } catch (e) {
+      const failureToast: Toast = {
+        id: 'update-result',
+        color: 'danger',
+        iconType: 'alert',
+        title: `Failed to update audit configuration due to ` + e?.message,
```

Comment:
Suggest use single quotes for plain string.

Reply:
Sure